switch/switch_main.ml

```diff
@@ -319,8 +319,14 @@ let main ({ Config.daemonize; path; pidfile } as config) =
         end else raise (Does_not_exist dir)
       | _ ->
         () );
+
     if daemonize
-    then Lwt_daemon.daemonize ();
+    then begin
+      info "Daemonizing...";
+      ignore (Daemon.notify Daemon.State.Ready);
+      Lwt_daemon.daemonize ();
```

Comment:
Do we really need to daemonise if this is started by systemd? Can't we can delete this entire if-statement if we use `Type=simple`?

Reply:
This also redirects the stdout/stdin. However, in `xapi-nbd` we just do this manually, and don't use `daemonize` at all: https://github.com/xapi-project/xapi-nbd/blob/master/src/main.ml#L212

Reply:
Can't we just let systemd deal with stdout/in/err?

Reply:
We did this in both the program and the `.service` file for `xapi-nbd`, for the sake of defense in depth: I think it's much safer that way. We've used the `.service` file as a last resort, in case we've forgotten to redirect something, or there are some errors that didn't get redirected:
```
StandardOutput=null
StandardError=syslog
```

Reply:
I think it's a good idea to get rid of this and rely only on systemd (consider also that Lwt_daemon is going to be deprecated in the next release). We could redirect stderr/stdout by hand instead.

Reply:
I think we need to `Daemon.notify` so systemd knows when message-switch is up and can start dependent services, but yes we can probably remove the rest of the code around daemonization.

Reply:
But what does "up" mean? I don't think it does anything relevant before sending the notify message. More useful would be to the systemd's socket activation in order to ensure that the socket is always ready.

Reply:
I think we only need to notify systemd if it's `Type=simple`. We ran into some issue with socket activation with `xapi-nbd` where after it entered the failed state, socket activation did not work.